src/inference/src/dev/property_manager.cpp

```diff
@@ -0,0 +1,183 @@
+
+    // Default global property will not be used for any plugin
+    ov::AnyMap default_global_properties;
+
+    // Internal properties
+    ov::AnyMap internal_properties;
+};
+
+void PropertyManagerImpl::merge_property(const ov::AnyMap& properties, const std::string& plugin_name) {
+    if (properties.size() == 0)
+        return;
+
+    std::lock_guard<std::mutex> lock(mutex);
+    auto merge = [&](const ov::AnyMap& src, ov::AnyMap& ref, ov::AnyMap& dst_a, ov::AnyMap& dst_b) {
+        for (auto& it : ref) {
+            auto item = src.find(it.first);
+            if (item != src.end()) {
+                dst_a[item->first] = item->second;
+                preprocess(dst_b, item->first, item->second.as<std::string>());
+            }
+        }
+    };
+    if (!plugin_name.empty()) {
+        ov::AnyMap value;
+        auto _value = plugin_properties.find(plugin_name);
+        if (_value != plugin_properties.end()) {
+            value = _value->second;
+        }
+        merge(properties, default_core_properties, value, value);
+        if (value.size() > 0)
+            plugin_properties[plugin_name] = value;
+    } else {
+        merge(properties, default_core_properties, default_core_properties, internal_properties);
+        merge(properties, default_global_properties, default_global_properties, internal_properties);
+    }
+}
+
+ov::AnyMap PropertyManagerImpl::exclude_property(const ov::AnyMap& property) {
+    auto res = property;
+
+    for (auto& it : default_core_properties) {
+        auto item = res.find(it.first);
+        if (item != res.end()) {
+            res.erase(item);
+        }
+    }
+    for (auto& it : default_global_properties) {
+        auto item = res.find(it.first);
+        if (item != res.end()) {
+            res.erase(item);
+        }
+    }
+    return res;
+}
+
+ov::Any PropertyManagerImpl::get_property(const std::string& name, const std::string& plugin_name) {
+    std::lock_guard<std::mutex> lock(mutex);
+    if (name == ov::force_tbb_terminate.name()) {
+        const auto flag = ov::threading::executor_manager()->get_property(name).as<bool>();
+        return flag ? CONFIG_VALUE(YES) : CONFIG_VALUE(NO);
+    }
+
+    if (!plugin_name.empty()) {
+        ov::AnyMap value;
+        auto _value = plugin_properties.find(plugin_name);
+        if (_value != plugin_properties.end()) {
+            value = _value->second;
+            if (value.count(name) > 0) {
+                return value.at(name);
+            }
+        }
+    }
+
+    if (default_core_properties.count(name) > 0) {
+        return default_core_properties.at(name);
+    }
+
+    if (default_global_properties.count(name) > 0) {
+        return default_global_properties.at(name);
+    }
+
+    if (internal_properties.count(name) > 0) {
+        return internal_properties.at(name);
+    }
+
+    if (name == ov::cache_manager.name()) {
+        std::shared_ptr<ov::ICacheManager> ret = nullptr;
+        return ov::Any(ret);
+    }
```

Comment:
why do we need to manage cache_manager explicitly? I suppose it's internal property, is not it?

Reply:
yes, it is internal property.
Here is just to return an empty pointer if caller want to get but it not set.
```
auto cacheManager = property_manager.get_property(ov::cache_manager.name(), plugin.get_name())
                            .as<std::shared_ptr<ov::ICacheManager>>();

if (cacheManager && device_supports_import_export(plugin)) {
```